p/scripts/main.js

```diff
@@ -692,21 +692,113 @@ function init_posts() {
+		return;
+	}
+
+	let open_categories = sessionStorage.getItem('FreshRSS_open_categories');
+	if (!open_categories) {
+		open_categories = '';
+	}
+
+	if (open_categories.indexOf(category_id) === -1) {
+		open_categories += category_id + ',';
+	}
+	sessionStorage.setItem('FreshRSS_open_categories', open_categories);
+}
+
+//Removes category id from session storage when user closes corresponding category in the left sidebar.
+function remove_category_from_session_storage(category_id) {
+	if ((category_id.charAt(0) !== 'c') && (category_id !== 'tags')) {
+		return;
+	}
+
+	let open_categories = sessionStorage.getItem('FreshRSS_open_categories');
+	if (open_categories.indexOf(category_id) !== -1) {
+		let re = new RegExp(`${category_id},`);
+		open_categories = open_categories.replace(re, '');
+	}
+	sessionStorage.setItem('FreshRSS_open_categories', open_categories);
+}
+
+//On page load, displays sidebar category dropdowns for categories opened by the user. 
+//Retrieves ids of categories from Session Storage.
+function display_open_categories() {
+	const open_categories = sessionStorage.getItem('FreshRSS_open_categories');
+	if (open_categories === null) {
+		//Page loaded for the first time. Drop down for category id already open.
+		//Add category id corresponding to URL 'get' parameter to session storage.
+		let request_item_id = get_search_param_value('get');
+		if (request_item_id && ['c', 'f', 'T', 't'].indexOf(request_item_id.charAt[0]) !== -1) {
+			let category_element = document.getElementById(request_item_id).closest('.category');
+			if (category_element) {
+				add_category_to_session_storage(category_element.id);
+			}
+		}
+	} else if (open_categories !== '') {
+		//Page loaded second time or later.
+		//Display category dropdowns in the UI for categories in open_categories.
+		let open_categories_array = open_categories.split(',');
+		open_categories_array.pop(); //Remove empty last element
+		open_categories_array.forEach(function(category_id) {			
+			let category_element = document.getElementById(category_id);
+			category_element.classList.add('active');
+			category_element.querySelector('.tree-folder-items').classList.add('active');
+			const img = category_element.querySelector('a.dropdown-toggle img');
+			img.src = img.src.replace('/icons/down.', '/icons/up.');
+			img.alt = '△';
+		});
+	}
+}
+
 function init_column_categories() {
 	if (context.current_view !== 'normal' && context.current_view !== 'reader') {
 		return;
 	}
 
+	display_open_categories();
+
+	// Set sidebar scroll position. Wait to let browser set the scroll position first.
+	setTimeout(function() { document.getElementById('sidebar').scrollTop = parseInt(sessionStorage.getItem('FreshRSS_sidebar_scroll_position')); }, 100);
+
+	// Sidebar scroll handler. Stores sidebar scroll position in Session Storage.
+	document.getElementById('sidebar').onscroll = function (event) {
+		sessionStorage.setItem('FreshRSS_sidebar_scroll_position', event.target.scrollTop);
```

Comment:
This looks very expensive to do.
In any case, I suggest removing the scroll position from this PR, and focus in this PR on saving the open/close state of the categories. The scroll position can be discussed in another issue / PR.

Reply:
Okay.